onedal/datatypes/data_conversion_dpctl.cpp

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifdef ONEDAL_DPCTL_INTEGRATION
+#define NO_IMPORT_ARRAY
+
+#include <stdexcept>
+#include <utility>
+#include <string>
+
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/detail/csr.hpp"
+#include "oneapi/dal/table/detail/homogen_utils.hpp"
+
+#include "onedal/datatypes/data_conversion_dpctl.hpp"
+#include "onedal/datatypes/numpy_helpers.hpp"
+
+#include "dpctl4pybind11.hpp"
+
+namespace oneapi::dal::python {
+
+void report_problem_from_dptensor(const char* clarification) {
+    constexpr const char* const base_message = "Unable to convert from dptensor";
+
+    std::string message{ base_message };
+    message += std::string{ clarification };
+    throw std::invalid_argument{ message };
+}
+
+std::int64_t get_and_check_dptensor_ndim(const dpctl::tensor::usm_ndarray& tensor) {
+    constexpr const char* const err_message = ": only 1D & 2D tensors are allowed";
```

Comment:
`: ` should be moved to `report_problem_from_dptensor`

Reply:
@Alexsandruss - I suppose functions like `report_error...` an be called without any conclusive issue